Source/DafnyCore/Resolver/PreType/PreType.cs

```diff
@@ -455,6 +455,9 @@ public class PreTypePlaceholderModule : PreTypePlaceholder {
   public class PreTypePlaceholderType : PreTypePlaceholder {
   }
 
+  /// Currently used exclusively for assigning a pre-type to MemberSelect expressions, such as "obj.method",
+  /// which is not considered an expression. This indicates that resolution has occurred,
```

Comment:
What do you mean by `MemberSelect` not being considered an expression? It inherits from `Expression`

Reply:
It weirdly is an Expression, but not a true Dafny expression: you cannot use it in a regular Dafny expression. Consider the following:

```
class A {
  method Test() {
  }
}
method Main() {
  var a := new A();
  var f := a.Test; // Error, expected a Dafny expression on the RHS
  f();
}
```

Reply:
It is also used for fields, in which case it is an expression, right?

Reply:
It seems nice to conceptually consider `MemberSelectExpr` that references a method, still an expression, even if we don't allow it in an expression context. If we do that, then I would expect to assign either a function type to it or a custom type `GenericCallableType`. Maybe we can rename `UnusedPreType` to `MethodType`

Reply:
Yes, it's used for fields and in that case it's indeed an expression.

I did the renaming, hoping it won't break any future development.